rfc/20250728-opentofu-providers/04-local-execution.md

```diff
@@ -0,0 +1,114 @@
+# Local Execution
+
+## Summary
+
+This document defines the configuration and execution model for running OpenTofu providers by provdiing a command and a set of arguments, enabling local development, package manager distribution, and flexible deployment patterns.
+
+## Configuration Model
+
+### Design Philosophy: Familiarity Over Innovation
+
+**Key Principle**: Use existing, familiar configuration structures rather than introducing new syntax. Users already understand the `required_providers` block from years of Terraform/OpenTofu usage. This RFC extends that familiar pattern rather than creating entirely new configuration mechanisms.
+
+### Relationship to Other Proposals
+
+This RFC is one of several approaches being explored for flexible provider execution, the combination of the two links below explore a different approach and is a recommended read before going into this document:
+
+- **[Local-exec providers (PR #3027)](https://github.com/opentofu/opentofu/pull/3027)**: Explores opt-in local execution with different configuration patterns
+- **[Registry in a file (PR #2892)](https://github.com/opentofu/opentofu/pull/2892)**: Proposes `.opentofu.deps.hcl` for source mapping and local overrides
+
+This RFC focuses on extending the existing `required_providers` syntax to be more flexible, while the other proposals introduce new configuration files or mechanisms. Each approach has trade-offs that should be discussed. In an ideal world we would find a way to introduce both approaches and this proposal does not block the concept of a Registry in a file.
+
+### Basic Configuration
+
+Providers can be configured using the familiar `required_providers` block, extended with `cmd` and `args` fields:
+
+```hcl
+terraform {
+  required_providers {
+    # Traditional registry-based provider (unchanged)
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+    
+    # Local script execution (new capability)
+    governance = {
+      cmd  = "python3"
+      args = ["./governance-provider.py", "--stdio"]
+    }
+    
+    # Package manager execution (new capability)
+    custom = {
+      cmd  = "npx"
+      args = ["-y", "@mycompany/provider@1.2.3", "--stdio"]
+    }
+
+    # Docker container execution (new capability)
+    scanner = {
```

Comment:
How would you pass this to a child module?  The child module will need *something* in their required_providers block, what would that look like?